File: contests/IX Maratona Mineira de Programação/mania.py
n = int(input())
s = input()
minv, maxv, pos = 0, 0, 0

for c in s:
  if c == 'E':
    pos -= n
    minv = min(minv, pos)
  else:
    pos += n
    maxv = max(maxv, pos)
    
if(abs(maxv) + abs(minv) >= 360):
  print("S")
else:
  print("N")

File: contests/IX Maratona Mineira de Programação/mania.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
  int n, pos, minv, maxv;
  string s;
  pos = minv = maxv = 0;
  cin >> n >> s;
  for(char c : s){
    if(c == 'E'){
      pos -= n;
      minv = min(minv, pos);
    }
    else{
      pos += n;
      maxv = max(maxv, pos);
    }
  }
  cout << (abs(maxv) + abs(minv) >= 360 ? "S" : "N") << endl;
  return 0;
}
